### PR TITLE
HAWQ-1004. Implement calling Ranger REST Service using libcurl for pr…

### PR DESCRIPTION
**Pull request by linwen**
Please review, thanks!

### CODE REVIEW

**Comment by zhangh43** [comment]
+1

**Comment by stanlyxiang** [comment]
+1

File: src/backend/libpq/rangerrest.c [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+    "oplass",            /* pg_opclass */
+    "conversion",        /* pg_conversion */
+    "tablespace",        /* pg_tablespace */
+    "filespace",         /* pg_filespace */
+    "filesystem",        /* pg_filesystem */
+    "fdw",               /* pg_foreign_data_wrapper */
+    "foreign_server",    /* pg_foreign_server */
+    "protocol",          /* pg_extprotocol */
+    "none"               /* MUST BE LAST */
+};
```
Comment (denalex) [inline src/backend/libpq/rangerrest.c]: a few of these are not supported in RPS/Ranger: operator, type, namespace, oplass, conversion, filespace, filesystem, fdw, foreign_server. Will these types be submitted to RPS or will they be mapped to other types ?

File: src/backend/libpq/rangerrest.c [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+  {
+    RangerRequestJsonArgs *arg_ptr = (RangerRequestJsonArgs *) lfirst(arg);
+    if (user == NULL)
+    {
+      user = arg_ptr->user;
+      juser = json_object_new_string(user);
+    }
+    AclObjectKind kind = arg_ptr->kind;
+    char* object = arg_ptr->object;
+    char* how = arg_ptr->how;
```
Comment (linwen) [inline src/backend/libpq/rangerrest.c]: Yes. We just leave it there, will modify it later.

**Pull request merged**